### ISSUE TITLE
GPR with stocastic gradient descent?

### ISSUE THREAD

**Issue by DallaLiberaAlberto** [GPflow/GPflow#730]
Hello,  first of all I wish to state that I am new with TensorFlow and GPflow. In my PhD I am working a lot with Gaussian Regression and I decided to adopt GPflow instead of my MATLAB implementation of GPR because it gives a lot of interesting functionalities. In particular I don't need to write explicitly the gradient of the kernel, and this is a great advantage, since I would use different semi-parametric and non-stationary kernels. I saw that there are several models and several training options. However it seems to me that standard GPR model (without latent inputs) don't support minibatches (I would like to adopt stochastic gradient descent techniques like Adam). Am I wrong?
P.S. I tried to modify the source code exploiting the Minibatch class in dataholder but I am not sure that it is working correctly

**Comment by mrksr** [comment]
Take a look at the SVGP model (and corresponding paper) - one of it's main features is that it supports minibatches. In the paper you can read why this is the case and maybe more importantly why it is not with other GP models.
